snapcraft/plugins/catkin.py

```diff
@@ -579,9 +579,10 @@ def _source_setup_sh(self, root, underlay_path):
             source_script = textwrap.dedent(
                 """
                 if [ -f {underlay_setup} ]; then
+                    set -- --local
                     _CATKIN_SETUP_DIR={underlay} . {underlay_setup}
                     if [ -f {rosdir_setup} ]; then
-                        set -- --extend
+                        set -- --local --extend
```

Comment:
Forgive my ignorance, but what is the effect of `set -- --<flag>`?  I would naively assume it as the shell's `set`, but that must not be?  I'm missing something of great value :thinking:

Reply:
`source` is a bashism that accepts parameters (e.g. `source <script> --flag`). Posix shell doesn't have that, and `.` doesn't support passing params, so this is a hack to get similar functionality by utilizing [set](https://pubs.opengroup.org/onlinepubs/009695399/utilities/set.html). We also [backup/restore](https://github.com/snapcore/snapcraft/pull/2847/files#diff-3cc26b49a70ee86523598a96dd385301R624-R625) any args passed to the ROS app as well so the script doesn't consume them instead.

Reply:
Oh wow, I have never seen `set` used to modify positional arguments.  TIL :wink: